loony_dev/git.py

```
@@ -189,3 +209,105 @@ def checkout_main(self) -> None:
+        without touching it.
+        """
+        if not branch.strip():
+            raise ValueError("branch must be non-empty")
+        resolved_base = base if base is not None else self.default_branch
+
+        for info in self.list_worktrees():
+            if info.path == path and info.branch == branch:
+                logger.debug("Worktree at %s already on %s; reusing", path, branch)
+                return path
+
+        subprocess.run(
```

Ruff (0.15.13)

[error] `subprocess` call: check for execution of untrusted input

(S603)